wemix/bind/backends/wemix_simulated_test.go

```diff
@@ -0,0 +1,175 @@
+
+	GetMaxBaseFee, err := contracts.EnvStorageImp.GetMaxBaseFee(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.MAX_BASE_FEE.Cmp(GetMaxBaseFee) == 0)
+
+	GetMaxIdleBlockInterval, err := contracts.EnvStorageImp.GetMaxIdleBlockInterval(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.MAX_IDLE_BLOCK_INTERVAL.Cmp(GetMaxIdleBlockInterval) == 0)
+
+	GetMaxPriorityFeePerGas, err := contracts.EnvStorageImp.GetMaxPriorityFeePerGas(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.MAX_PRIORITY_FEE_PER_GAS.Cmp(GetMaxPriorityFeePerGas) == 0)
+
+	GetStakingMax, err := contracts.EnvStorageImp.GetStakingMax(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.STAKING_MAX.Cmp(GetStakingMax) == 0)
+
+	GetStakingMin, err := contracts.EnvStorageImp.GetStakingMin(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.STAKING_MIN.Cmp(GetStakingMin) == 0)
+
+	getNode, err := contracts.GovImp.GetNode(callOpts, common.Big1)
+	require.NoError(t, err)
+	t.Log(
+		"\n Name:", string(getNode.Name),
+		"\n Enode:", hex.EncodeToString(getNode.Enode),
+		"\n Ip:", string(getNode.Ip),
+		"\n Port:", getNode.Port,
+	)
+	newVoterKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	newVoter := crypto.PubkeyToAddress(newVoterKey.PublicKey)
+
+	LOCK_AMOUNT, err := contracts.EnvStorageImp.GetStakingMax(callOpts)
+	require.NoError(t, err)
+
+	tx, err := contracts.GovImp.AddProposalToChangeMember(opts, gov.GovImpMemberInfo{
+		Staker:     opts.From,
+		Voter:      newVoter,
+		Reward:     opts.From,
+		Name:       getNode.Name,
+		Enode:      getNode.Enode,
+		Ip:         getNode.Ip,
+		Port:       getNode.Port,
+		LockAmount: LOCK_AMOUNT,
+		Memo:       []byte("memo1"),
+		Duration:   big.NewInt(86400),
+	}, opts.From, common.Big0, common.Big0)
+	require.NoError(t, err)
+
+	receipt, err := bind.WaitMined(context.Background(), b, tx)
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+
+	inVoting, err := contracts.GovImp.GetBallotInVoting(callOpts)
+	require.NoError(t, err)
+	require.True(t, inVoting.Sign() == 0)
+
+	member, err := contracts.GovImp.GetMember(callOpts, common.Big1)
+	require.NoError(t, err)
+	require.Equal(t, opts.From, member)
+
+	voter, err := contracts.GovImp.GetVoter(callOpts, common.Big1)
+	require.NoError(t, err)
+	require.Equal(t, newVoter, voter)
+
+	newReward, err := contracts.GovImp.GetReward(callOpts, common.Big1)
+	require.NoError(t, err)
+	require.Equal(t, opts.From, newReward)
+
+	newHead := make(chan *types.Header)
+	sub, err := b.SubscribeNewHead(context.Background(), newHead)
+	require.NoError(t, err)
+
+	ticker := time.NewTicker(10e9)
+	var headNumber *big.Int = nil
+loop:
+	for {
+		b.Commit()
+		select {
+		case head := <-newHead:
+			if headNumber == nil {
+				headNumber = head.Number
+			} else {
+				require.Equal(t, new(big.Int).Add(headNumber, common.Big1), head.Number)
+				headNumber = head.Number
+			}
+			time.Sleep(0.2e9)
```

Comment:
Is delay processing for the newHeader message necessary?

Reply:
removed as it was unnecessary.